### PR TITLE
Remove regex and fnmatch again on NodeNameResolver

### PR DESCRIPTION
**Pull request by samsonasik**
Per @TomasVotruba request https://github.com/rectorphp/rector-src/pull/4964#issuecomment-1712814835

### CODE REVIEW

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba @staabm I am merging it ;)

**Comment by samsonasik** [comment]
@TomasVotruba for note: this need be noted bold as BC break in next rector release :)

**Comment by TomasVotruba** [comment]
Thanks :pray:

File: rules/Php55/Rector/String_/StringClassNameToClassConstantRector.php [hunk 1]
```diff
@@ -179,6 +179,10 @@ private function shouldSkip(string $classLikeName): bool
             if ($this->nodeNameResolver->isStringName($classLikeName, $classToSkip)) {
                 return true;
             }
+
+            if (fnmatch($classToSkip, $classLikeName, FNM_NOESCAPE)) {
```
Comment (staabm) [inline rules/Php55/Rector/String_/StringClassNameToClassConstantRector.php:183]: Maybe we can throw when wildcards are contained to assist the user in the upgrade path?

Comment (TomasVotruba) [reply rules/Php55/Rector/String_/StringClassNameToClassConstantRector.php:183]: Here it's actually expected, as both formats are supported. 

But we could inform in the `isName()` method to inform about BC break :+1:

Comment (staabm) [reply rules/Php55/Rector/String_/StringClassNameToClassConstantRector.php:183]: Thats what I had in mind, right :-)

**Pull request merged**